modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy

```diff
@@ -115,6 +115,20 @@ def buildProject(Map options = [:]) {
     [$class: 'ParametersDefinitionProperty', parameterDefinitions: parameterDefinitions],
   ])
 
+  def defaultParameterValuesMap = [:]
+  parameterDefinitions.each {
+    // to handle params defined with the xxxParam(...) DSL instead of
+    // [$class: ... ] style because we can't call .name / .defaultValue
+    // on them directly
+    if (it.class == org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable) {
+      def mapVersionOfIt = it.toMap()
```

Comment:
Instead of using `toMap` (if it means we have to enable something in Jenkins), can we use [`getArguments`](http://javadoc.jenkins-ci.org/structs/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.html#getArguments--) or [`instantiate`](http://javadoc.jenkins-ci.org/structs/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.html#instantiate--) ?

Reply:
Oh, good idea, I'll try it and see what happens, but I think the groovy sandbox is very limited.

Reply:
Alas, same problem with `getArguments`:

[`org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable getArguments`](https://ci.integration.publishing.service.gov.uk/job/publishing-api/job/use_rabbitmq_envvar/16/console)

and `instantiate`:

[`org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable instantiate`](https://ci.integration.publishing.service.gov.uk/job/publishing-api/job/use_rabbitmq_envvar/17/console).

Shame.